### PR TITLE
Building for all platforms, to merge into main

### PR DESCRIPTION
**Pull request by Daresoul**

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  pull_request:
+    branches: ["main"]
+    types: [opened, synchronize]
+
+jobs:
+  build-tauri:
+    permissions:
+      contents: write
+    strategy:
+      fail-fast: false
```
Comment (leny-mi) [inline .github/workflows/main.yml:13]: Why do we not fail-fast?

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+          sudo apt-get install -y libgtk-3-dev libwebkit2gtk-4.0-dev libappindicator3-dev librsvg2-dev patchelf
+      - name: install app dependencies
+        run: yarn
+      - name: build the app
+        run: npx turbo build
+      - uses: tauri-apps/tauri-action@v0
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          releaseId: ${{ needs.create-release.outputs.release_id }}
```
Comment (leny-mi) [inline .github/workflows/main.yml:39]: Why does tauri-action need a releaseId? I thought this is just a smoke test?

Comment (Daresoul) [reply .github/workflows/main.yml:39]: An oversight on my end, from when the release was part of this scope.

File: .github/workflows/main.yml (hunk 1 above)
Comment (Daresoul) [reply .github/workflows/main.yml:13]: Agreed with your other comment, should be a smoke test, can it give value to see all platforms that fail, but will agree is not that necesarry for the smoketest.

**Review by Daresoul — COMMENTED** [review]
Will implement from the comments, am gonna set fail-fast to true.

File: .github/workflows/main.yml (hunk 2 above)
Comment (Daresoul) [reply .github/workflows/main.yml:39]: It is something you can optionally give to it.

**Review by leny-mi — APPROVED** [review]
Lgtm!

**Pull request merged**